### PR TITLE
Fix no-ec (again)

### PR DESCRIPTION
**Pull request by mattcaswell**
We broke no-ec again. I only just fixed it this morning :-)

This PR is mainly about fixing no-sm2, and then if no-ec is requested then this implies no-sm2.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: test/evp_test.c [hunk 1]
```diff
@@ -2413,6 +2413,23 @@ static char *take_value(PAIR *pp)
     return p;
 }
 
+static int key_disabled(EVP_PKEY *pkey)
+{
+#if defined(OPENSSL_NO_SM2) && !defined(OPENSSL_NO_EC)
```
Comment (levitte) [inline test/evp_test.c]: I think this line could do with a little bit more bang (!)

Comment (mattcaswell) [reply test/evp_test.c]: No - this is correct. If we have asked for sm2 to be disabled, but ec is still enabled then we may end up reading ec keys with an sm2 group. We need to detect those and ignore them.

**Comment by tmshort** [comment]
I was about to make a similar fix

**Comment by tmshort** [comment]
I have some additional fixes related to SM2 and no-sm3.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

> I have some additional fixes related to SM2 and no-sm3.

Great. Anything urgent, or can it wait until after the beta?

File: test/evp_test.c (hunk 1 above)
Comment (tmshort) [reply test/evp_test.c]: Would an easier solution be to not add the NID_sm2 to the `curve_list[]` in ec_curve.c?

**Comment by tmshort** [comment]
I think this should be closed?

**Comment by mattcaswell** [comment]
Oh, right! Yes - thanks. This was merged.

**Pull request closed without merging**